pkg/resolvers/k8s/resolver_test.go

```diff
@@ -68,17 +68,19 @@ func TestParseTarget(t *testing.T) {
 				},
 			},
 		},
-		{
-			target: "service5.ns5:some-port",
-			expectgedTarget: targetEntry{
-				service:   "service5",
-				namespace: "ns5",
-				port: targetPort{
-					value:   "some-port",
-					isNamed: true,
-				},
-			},
-		},
+		// After Go 1.12 non-numeric ports are not accepted by net/url.Parse anymore.
+		// See https://github.com/golang/go/issues/33646, https://play.golang.org/p/5KK1Qdy05b
+		//{
+		//	target: "service5.ns5:some-port",
+		//	expectgedTarget: targetEntry{
+		//		service:   "service5",
+		//		namespace: "ns5",
+		//		port: targetPort{
+		//			value:   "some-port",
+		//			isNamed: true,
+		//		},
+		//	},
+		//},
```

Comment:
Please don't comment out code, just delete it. In this case, you'll also need to update the parser in resolver.go to remove the support for named ports.

Reply:
Fair comment, I mostly left this here for discusson, will nuke it now if we are happy with it.
I will have a look at resolver.go.